activerecord/lib/active_record/core.rb

```diff
@@ -162,9 +162,16 @@ def initialize_find_by_cache # :nodoc:
         @find_by_statement_cache = { true => Concurrent::Map.new, false => Concurrent::Map.new }
       end
 
+      def define_table_constant(child_class)
+        # We have a connection at this point, which we need for this to work
```

Comment:
I don't _think_ we do have a connection at this point, but I also don't think we do need it 😅

Reply:
This PR is a little out of date at this point - I have done some work locally here and this doesn't reflect what we have. 

I think we do need a connection in order to infer the column names though.

Reply:
👍 It's fine if we need a connection to do `MyModel::Table.name`, but we can't expect it to be connected when defining the model class.